test/e2e/autotls/auto_tls_test.go

```diff
@@ -63,65 +91,51 @@ func TestPerKsvcCertLocalCA(t *testing.T) {
 		t.Fatalf("Failed to create initial Service: %v: %v", names.Service, err)
 	}
 
-	cancel := turnOnAutoTLS(t, clients)
-	test.CleanupOnInterrupt(cancel)
-	defer cancel()
-
-	// Wait for certificate to be ready.
-	if err := waitForCertificateReady(t, clients, routenames.Certificate(objects.Route)); err != nil {
-		t.Fatalf("Error waiting for certificate to become ready: %v", err)
+	certName := getCertificateName(t, clients, objects)
+	if err := waitForCertificateReady(t, clients, certName); err != nil {
```

Comment:
See my comment below about the `https://` scheme, but it feels like the focus of this test should be less on the `Certificate` mechanics and more on the manifestation in `ksvc`.

Feels like we should be able to wait for the `ksvc` to have an `https://` scheme URL and curl that (possibly with different root CAs).

At this level, do we even need to much with Certs?  Seems like you'd shed a bunch of gnarly logic with that simplification which feels closer to what users would see black boxing things, right?